src/isl/isl_type_reference.rs

```diff
@@ -47,13 +72,25 @@ impl IslTypeRef {
                         "a base or alias type reference can not be null.symbol",
                     )
                 }
-                value.as_sym().unwrap()
+
+                let type_name = value.as_sym().unwrap()
                     .text()
                     .ok_or_else(|| {
                         invalid_schema_error_raw(
                             "a base or alias type reference symbol doesn't have text",
                         )
-                    }).map(|type_name| IslTypeRef::Named(type_name.to_owned()))
+                    })?;
+
+                // check for nullable type reference
+                if value.annotations().any(|a| a == &text_token("nullable")) {
+                    let built_in_type_def = IslTypeRef::get_nullable_type_reference_definition(type_name)?;
+                    let value = &element_reader()
+                        .read_one(built_in_type_def.as_bytes())
+                        .map_err(|_| invalid_schema_error_raw(format!("Can not parse nullable type reference for given built in type: {}", type_name)))?;
+                    return IslTypeRef::from_ion_element(value, inline_imported_types);
```

Comment:
Not a blocker: it seems like we should be able to cache these instead of parsing/building them each time they're referenced in the schema.

Reply:
Sounds good! I would just add the `OwnedElement` for them directly.

Reply:
Reference issue: https://github.com/amzn/ion-schema-rust/issues/101